src/lib.rs

```diff
@@ -174,12 +175,13 @@ pub fn run(mut vm: VMState) -> (U256, VMState) {
                 },
             }
         }
-        vm.current_context_mut().pc += 1;
-        vm.decrease_gas(&opcode);
+        vm.current_frame_mut().pc += 1;
     }
-    let final_storage_value = match vm.current_context().storage.borrow().read(&U256::zero()) {
-        Ok(value) => value,
-        Err(_) => U256::zero(),
-    };
+    let final_storage_value = vm
+        .current_frame()
+        .storage
+        .borrow()
+        .read(&U256::zero())
+        .unwrap();
```

Comment:
Is there a reason why we changed the match clause? it doesn't matter that much but just out of curiosity.

Reply:
The change of `current_context` for `current_frame` was intentional, since now the `current_context` is actually just the one of the far calls, and current frame could be from a near call.
About the match for the unwrap, it was not...
However I think it's better this way, since a lot of times, some tests failed because they didn't reach the `sstore` instruction, and having the unwrap was a really quick way of knowing why it failed, if it just returned 0, then It would have been more difficult to notice.
So in the end, it was not on purpose, but useful anyways